File: Knjigoteka.Model/Entities/RestockRequest.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Knjigoteka.Model.Entities
{
    public class RestockRequest
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int BookId { get; set; }
        [Required]
        public int BranchId { get; set; }
        [Required]
        public int EmployeeId { get; set; }
        [Required]
        public DateTime RequestDate { get; set; } = DateTime.Now;

        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "Quantity must be greater than 0.")]
        public int QuantityRequested { get; set; }
        [Required]
        public RestockRequestStatus Status { get; set; } = RestockRequestStatus.Pending;

        [ForeignKey(nameof(BookId))]
        public Book Book { get; set; } = null!;
        [ForeignKey(nameof(BranchId))]
        public Branch Branch { get; set; } = null!;
        [ForeignKey(nameof(EmployeeId))]
        public Employee Employee { get; set; } = null!;
    }
    public enum RestockRequestStatus
    {
        Pending,
        Approved,
        Recieved,
        Rejected
    }

}

File: Knjigoteka.UI/knjigoteka_desktop/lib/models/restock_request.dart
enum RestockRequestStatus { pending, approved, rejected }

RestockRequestStatus StatusFromInt(int val) {
  switch (val) {
    case 1:
      return RestockRequestStatus.approved;
    case 2:
      return RestockRequestStatus.rejected;
    default:
      return RestockRequestStatus.pending;
  }
}

class RestockRequest {
  final int id;
  final int bookId;
  final String bookTitle;
  final int branchId;
  final String branchName;
  final int employeeId;
  final String employeeName;
  final DateTime requestedAt;
  final int quantityRequested;
  final RestockRequestStatus status;

  RestockRequest({
    required this.id,
    required this.bookId,
    required this.bookTitle,
    required this.branchId,
    required this.branchName,
    required this.employeeId,
    required this.employeeName,
    required this.requestedAt,
    required this.quantityRequested,
    required this.status,
  });

  factory RestockRequest.fromJson(Map<String, dynamic> json) {
    return RestockRequest(
      id: json['id'],
      bookId: json['bookId'],
      bookTitle: json['bookTitle'],
      branchId: json['branchId'],
      branchName: json['branchName'],
      employeeId: json['employeeId'],
      employeeName: json['employeeName'],
      requestedAt: DateTime.parse(json['requestedAt']),
      quantityRequested: json['quantityRequested'],
      status: StatusFromInt(json['status']),
    );
  }
}
